sharpd/sharp_vty.c

```diff
@@ -337,11 +337,137 @@ DEFUN_NOSH (show_debugging_sharpd,
 	    DEBUG_STR
 	    "Sharp Information\n")
 {
-	vty_out(vty, "Sharp debugging status\n");
+	vty_out(vty, "Sharp debugging status:\n");
 
 	return CMD_SUCCESS;
 }
 
+DEFPY (sharp_lsp_prefix_v4, sharp_lsp_prefix_v4_cmd,
+       "sharp lsp (0-100000)$inlabel\
+        nexthop-group NHGNAME$nhgname\
+        [prefix A.B.C.D/M$pfx\
+       " FRR_IP_REDIST_STR_SHARPD "$type_str [instance (0-255)$instance]]",
+       "Sharp Routing Protocol\n"
+       "Add an LSP\n"
+       "The ingress label to use\n"
+       "Use nexthops from a nexthop-group\n"
+       "The nexthop-group name\n"
+       "Label a prefix\n"
+       "The v4 prefix to label\n"
+       FRR_IP_REDIST_HELP_STR_SHARPD
+       "Instance to use\n"
+       "Instance\n")
+{
+	struct nexthop_group_cmd *nhgc = NULL;
+	struct prefix p = {};
+	int type = 0;
+
+	/* We're offered a v4 prefix */
+	if (pfx->prefixlen > 0 && type_str) {
```

Comment:
I believe `pfx` itself will be null if not provided?

http://docs.frrouting.org/projects/dev-guide/en/latest/cli.html#type-rules

Reply:
No, what I get is a prefix struct that's all zeroes - not a NULL pointer. but if it'd be clearer to test `pfx` before dereferencing it, that'd be fine, if it's distracting?

Reply:
Okay, then we need to update the docs:

![2020-02-18-113726_851x296_scrot](https://user-images.githubusercontent.com/6827003/74756752-13ff3400-5243-11ea-8ae0-4c581fec6411.png)

Reply:
(Obviously there is no point in testing a known good ptr)

Reply:
eva# sharp lsp 393 nexthop-group A prefix 0.0.0.0/0 kernel 
%  Must supply both prefix and type
eva# 

it sure seems like I passed a valid prefix in.